velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#define XXH_INLINE_ALL
+
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+#include "velox/external/xxhash.h"
+
+namespace facebook::velox::aggregate {
+template <TypeKind kind>
+int64_t getHash(const DecodedVector& vector, const vector_size_t row) {
+  return getHash<kind>(vector, row);
```

Comment:
Would `getHash` call itself in this general case?

Reply:
It should switch to the instantiated version, but this code isnt required. let me remove it.